test/test-changelist.c

```diff
@@ -228,6 +228,10 @@ main(int argc, char **argv)
 		event_free(ev);
 	if (timeout)
 		event_free(timeout);
+	if (pair[0] >= 0)
```

Comment:
It is not documented does evutil_socketpair will fill the pair in case of failure or not, so you need to initialize it with `EVUTIL_INVALID_SOCKET` before calling and then check against it, and not `>=0`